File: frontend/src/types/product.ts
/** Mirrors aio.models.product -- the Product Manager's typed output. */

export interface UserStory {
  as_a: string;
  i_want: string;
  so_that: string;
  acceptance_criteria: string[];
}

export interface Epic {
  title: string;
  description: string;
  user_stories: UserStory[];
}

export interface ProductVision {
  statement: string;
  target_users: string[];
  value_proposition: string;
}

export interface Risk {
  description: string;
  likelihood: string;
  impact: string;
  mitigation: string;
}

export interface SuccessMetric {
  name: string;
  target: string;
  rationale: string;
}

export interface ReleasePhase {
  name: string;
  scope: string;
  epics: string[];
}

export interface BusinessRequirementsDocument {
  vision: ProductVision;
  epics: Epic[];
  release_roadmap: ReleasePhase[];
  sprint_suggestions: string[];
  risk_register: Risk[];
  success_metrics: SuccessMetric[];
  confidence: number;
  reasoning_summary: string;
}

File: src/aio/models/product.py
"""Structured output for the Product Manager once it's research-driven."""

from __future__ import annotations

from pydantic import BaseModel, Field


class UserStory(BaseModel):
    as_a: str
    i_want: str
    so_that: str
    acceptance_criteria: list[str] = Field(default_factory=list)


class Epic(BaseModel):
    title: str
    description: str
    user_stories: list[UserStory] = Field(default_factory=list)


class ProductVision(BaseModel):
    statement: str
    target_users: list[str] = Field(default_factory=list)
    value_proposition: str


class Risk(BaseModel):
    description: str
    likelihood: str  # "low" | "medium" | "high"
    impact: str  # "low" | "medium" | "high"
    mitigation: str


class SuccessMetric(BaseModel):
    name: str
    target: str
    rationale: str


class ReleasePhase(BaseModel):
    name: str
    scope: str
    epics: list[str] = Field(default_factory=list)


class BusinessRequirementsDocument(BaseModel):
    vision: ProductVision
    epics: list[Epic] = Field(default_factory=list)
    release_roadmap: list[ReleasePhase] = Field(default_factory=list)
    sprint_suggestions: list[str] = Field(default_factory=list)
    risk_register: list[Risk] = Field(default_factory=list)
    success_metrics: list[SuccessMetric] = Field(default_factory=list)
    confidence: float = Field(ge=0.0, le=1.0)
    reasoning_summary: str
